pkg/network/usm/ebpf_gotls.go

```diff
@@ -320,6 +320,10 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
 }
 
 func (p *goTLSProgram) handleProcessStart(pid pid) {
+	if p.cfg.GoTLSExcludeSelf && pid == uint32(os.Getpid()) {
```

Comment:
Where is `pid` coming from? I ask because sometimes the containers run in separate PID namespaces. If `pid` comes from the kernel and is from the root namespace, you need to use something like `manager.Getpid()` to get the non-namespaced PID for system-probe.

Reply:
If you want to test whether your logic is correct, run system-probe in a docker container (which uses PID namespaces by default).